tools/backup_analysis.py

```diff
@@ -0,0 +1,174 @@
+    start = time.asctime()
+    # Compare the bytes in cmp_size blocks between both files.
+    for b in range(0, num_blocks + 1):
+        offset = cmp_size * b
+        # Gather and report information when we cross a granularity boundary.
+        if offset % granularity == 0:
+            if offset != 0 and bytes_gran != 0:
+                print("{}: Offset {}: {} bytes differ in {} bytes".format(filename, start_off, bytes_gran, granularity))
+            # Account for small or large block changes.
+            if bytes_gran != 0:
+                if bytes_gran <= pct20:
+                    pct20_count += 1
+                elif bytes_gran >= pct80:
+                    pct80_count += 1
+            # Reset for the next granularity block.
+            start_off = offset
+            bytes_gran = 0
+
+        # Compare the two blocks. We know both files are at least file_size so all reads should work.
+        buf1 = fp1.read(cmp_size)
+        buf2 = fp2.read(cmp_size)
+        # If they're different, gather information.
+        if buf1 != buf2:
+            total_bytes_diff += cmp_size
+            # Count how many granularity level blocks changed.
+            if bytes_gran == 0:
+                gran_blocks += 1
+            bytes_gran += cmp_size
+    # Account for anything from the final iteration of the loop and partial blocks.
+    if partial != 0:
+        buf1 = fp1.read(partial)
+        buf2 = fp2.read(partial)
+        # If they're different, gather information.
+        if buf1 != buf2:
+            total_bytes_diff += partial
+            bytes_gran += partial
+    fp1.close()
+    fp2.close()
+    end = time.asctime()
+    if bytes_gran != 0:
+        if offset != 0:
+            print("{}: Offset {}: {} bytes differ in {} bytes".format(filename, start_off, bytes_gran, granularity))
+        if bytes_gran <= pct20:
+            pct20_count += 1
+        elif bytes_gran >= pct80:
+            pct80_count += 1
+
+    # Report for each file.
+    print("{}: started {} completed {}".format(filename, start, end))
+    print("{} ({}) differs by {} bytes in {} granularity blocks".format(filename, file_size, total_bytes_diff, gran_blocks ))
+    if gran_blocks != 0:
+        pct20_blocks = round(abs(pct20_count / gran_blocks * 100))
+        pct80_blocks = round(abs(pct80_count / gran_blocks * 100))
+        print("{}: {} of {} blocks ({}%) differ by {} bytes or less of {} (20%)".format(filename, pct20_count, gran_blocks, pct20_blocks, pct20, granularity))
+        print("{}: {} of {} blocks ({}%) differ by {} bytes or more of {} (80%)".format(filename, pct80_count, gran_blocks, pct80_blocks, pct80, granularity))
+
+def compare_backups(dir1, dir2, granularity):
+    common=set()
+    files1=set(fnmatch.filter(os.listdir(dir1), "*.wt"))
+    files2=set(fnmatch.filter(os.listdir(dir2), "*.wt"))
+
+    common = files1.intersection(files2)
+    # For now assume the first directory is the older one. Once we add functionality to parse the
```

Comment:
You may want to include this assumption in the usage info.

Reply:
Done.

Reply:
I also added another line to remind the developer here to update the usage info when that work is done.